Codecademy/discourse-codebytes-plugin
# Have window message handlers check the type of event

## Overview
Ran into a bug where saves were being triggered at incorrect times due to other window messages. This PR updates the message handler here to check for the `codeBytesSaveResponse` key. It also makes the message sent to the child iframe more explicit by setting `codeBytesSaveRequested`.

### PR Checklist

### Notes
Depends on https://github.com/codecademy-engineering/static-sites/pull/663. 

### Testing Instructions
1. Static sites: `git checkout dd-explicit-message-events` then `yarn develop`
2. Run Discourse with bundle exec rails server in this directory
3. Visit http://localhost:3000/
4. Click the "New Topic" button to start a new post
5. Click the "C" button to add a new CodeByte
6. Make some changes to the codebyte in the markdown editor on the left. Check that the new code appears in the codebyte editor on the right.
7. Make some changes to the codebyte in the codebyte editor on the right. Click the "Save to post" button. Check that the new code appears in the markdown editor on the left.
8. Repeat the above steps on a topic with multiple codebytes. Make sure that only the one currently being edited is changed.
